src/scatter_buildblock/ScatterSimulation.cxx

```diff
@@ -959,8 +959,7 @@ ScatterSimulation::downsample_scanner(int new_num_rings, int new_num_dets)
     
     const float approx_num_non_arccorrected_bins =
       this->proj_data_info_sptr->get_num_tangential_poss() * 
-      (float(new_num_dets) / old_scanner_ptr->get_num_detectors_per_ring())
-      + 5; // add 5 to avoid strange edge-effects, certainly with B-splines
```

Comment:
@KrisThielemans: I have removed this here because it caused the check for #tangential bins in interpolate_projdata to fail. Since I am now extending the sinograms not only in views, but also the other two directions during the interpolation of Cylindrical scanners I don't think this is no longer needed here. Do you agree?

Reply:
hmmm. `extend` is a guess,/extrapolation while increasing the number of bins that is computed is at the expense of computation time but increases accuracy. I'd rather have it slightly higher, although 5 is now possibly too much. what about using `ceil(approx_num_non_arccorrected_bins)+1`?

which check failed though? That seems a bug in itself.

Reply:
https://github.com/markus-jehl/STIR/blob/d9aa91bcd260efcf57152a40becd4770eff4a148/src/buildblock/interpolate_projdata.cxx#L261

This one, which we just added to confirm that we can map data across directly. What I could do instead is confirm that the downsampled bins are at least as many as the upsampled ones.

Reply:
Not sure what your `sample_function_using_index_converter` will do then, but it looks fine.

But it would make sense to only add 5 (or 2) if there is actually any downsampling going on. Otherwise we compute stuff to throw it away.

Reply:
That's true, this is definitely the best approach. Will do that